### PR TITLE
Tn 32 make tokens jwt

### PR DESCRIPTION
**Pull request by joaov2**

### CODE REVIEW

File: app/repositories/slick/implementations/SlickAuthenticationRepository.scala [hunk 1]
```diff
@@ -129,13 +137,17 @@ class SlickAuthenticationRepository @Inject() (db: Database)(implicit executionC
...
+      if (_) {
+        JwtJson.decodeJson(token, key, Seq(algo)) match {
+          case Success(json) => json.validate[Jwt].fold(
+            errors => Left(internalError),
+            jwt => if (jwt.expirationDate.isAfter(LocalDateTime.now))
+              Right(jwt.userId)
+            else Left(tokenNotValid))
+          case Failure(e) => Left(internalError)
+        }
+      } else Left(tokenNotValid))
```
Comment (IvoSaraiva) [inline app/repositories/slick/implementations/SlickAuthenticationRepository.scala]: You should check expiration before going to the db. That was the purpose of putting the date in the token.

File: test/repositories/slick/implementations/SlickAuthenticationRepositorySpec.scala [hunk 2]
```diff
@@ -80,21 +88,36 @@ class SlickAuthenticationRepositorySpec extends AsyncWordSpec
...
-      testQuery.map {
-        case Some(tuple) => assert(tuple._1 === userAccess.address &&
-          tuple._2 === userAccess.password && tuple._3 === userAccess.first_name.value &&
-          tuple._4 === userAccess.last_name.value && DataValidators.isValidUUID(tuple._5))
+      val userAccess: UserAccess = genUserAccess.sample.value
+      val key: String = config.get[String]("secretKey")
+      val algo = JwtAlgorithm.HS256
+
+      for {
+        createdtoken <- authenticationRep.signUpUser(userAccess)
```
Comment (IvoSaraiva) [inline test/repositories/slick/implementations/SlickAuthenticationRepositorySpec.scala]: signUp or signIn?

File: test/repositories/slick/implementations/SlickAuthenticationRepositorySpec.scala [hunk 3]
```diff
@@ -145,43 +188,69 @@ class SlickAuthenticationRepositorySpec extends AsyncWordSpec
...
-    "get User" in {
+    "detect failure to validate the token" in {
       val tokenId = genUUID.sample.value
       val token = genUUID.sample.value
-      val passwordId = genUUID.sample.value
-      val userId = genUUID.sample.value
-      val password = genString.sample.value
+      for {
+        _ <- db.run(DBIO.seq(
+          TokensTable.all += TokenRow(tokenId, token)))
```
Comment (IvoSaraiva) [inline test/repositories/slick/implementations/SlickAuthenticationRepositorySpec.scala]: Why the DBIO.seq?

File: test/repositories/slick/implementations/SlickAuthenticationRepositorySpec.scala [hunk 4]
```diff
@@ -145,43 +188,69 @@ class SlickAuthenticationRepositorySpec extends AsyncWordSpec
...
+      val algo = JwtAlgorithm.HS256
+
+      val claim = Json.obj(("userId", genUUID.sample.value), ("expirationDate", LocalDateTime.now.minusDays(1)))
+      val token = JwtJson.encode(claim, key, algo)
 
       for {
         _ <- db.run(DBIO.seq(
-          TokensTable.all += TokenRow(tokenId, token, currentTimestamp, currentTimestamp),
-          PasswordsTable.all += PasswordRow(passwordId, userId, password, tokenId)))
+          TokensTable.all += TokenRow(tokenId, token)))
```
Comment (IvoSaraiva) [inline test/repositories/slick/implementations/SlickAuthenticationRepositorySpec.scala]: Why the DBIO.seq?

File: test/repositories/slick/implementations/SlickAuthenticationRepositorySpec.scala (hunk 2 above)
Comment (joaov2) [reply test/repositories/slick/implementations/SlickAuthenticationRepositorySpec.scala]: SignUp, that test is testing the signUp method. The repository no longer has a signIn method. The closest thing to it is the updateToken method.

**Pull request merged**